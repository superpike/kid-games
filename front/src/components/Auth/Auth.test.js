import React from 'react';
import { render } from '@testing-library/react';
import { Auth } from './index';
import '@testing-library/jest-dom/extend-expect';

describe('Testing Auth component', () => {
  const username = 'test user';
  const fakeLogin = jest.fn(data => data);
  const fakeRegister = jest.fn(data => data);
  const fakeLogout = jest.fn(data => data);

  describe('Is matched to snapshots', () => {
    test('should match to unlogged snapshot', () => {
      const { container } = render(<Auth />);
      expect(container).toMatchSnapshot();
    });
    test('should match to logged snapshot', () => {
      const { container } = render(<Auth username={username} />);
      expect(container).toMatchSnapshot();
    });
  });
  describe('Buttons clicks', () => {
    test('should start login process', () => {
      const { getByTestId } = render(
        <Auth login={fakeLogin} register={fakeRegister} />
      );
      const button = getByTestId('login');
      button.click();
      expect(fakeLogin).toHaveBeenCalledTimes(1);
    });
    test('should start register process', () => {
      const { container } = render(
        <Auth login={fakeLogin} register={fakeRegister} />
      );
      const button = getByTestId('register');
      button.click();
      expect(fakeRegister).toHaveBeenCalledTimes(1);
    });
    test('should start logout process', () => {
      const { container } = render(
        <Auth username={username} logout={fakeLogout} />
      );
      const button = getByTestId('logout');
      button.click();
      expect(fakeLogout).toHaveBeenCalledTimes(1);
    });
  });
});
