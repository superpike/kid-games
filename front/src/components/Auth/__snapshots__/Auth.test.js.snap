// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Auth component Is matched to snapshots should match to logged snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div>
      test user
    </div>
    <button
      class="Danger Button"
      data-testid="logout"
      type="button"
    >
      logout
    </button>
  </div>
</div>
`;

exports[`Testing Auth component Is matched to snapshots should match to unlogged snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <button
      class="Auth Button"
      data-testid="register"
      type="button"
    >
      register
    </button>
    <button
      class="Auth Button"
      data-testid="login"
      type="button"
    >
      login
    </button>
  </div>
</div>
`;
